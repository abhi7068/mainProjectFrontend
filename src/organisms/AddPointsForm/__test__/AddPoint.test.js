import React from 'react';
import ReactDOM from 'react-dom';
import Addpoint from '../index';
import { render } from '@testing-library/react';

it('it renders', () => {
  render(<Addpoint />);
});
it('should take a snapshot', () => {
  const { asFragment } = render(<Addpoint />);
  expect(asFragment(render(<Addpoint />))).toMatchSnapshot();
});
it('it renders with correct text', () => {
  const { getByTestId } = render(<Addpoint />);
  expect(getByTestId('simple')).toHaveTextContent('Add Points');
});
it('renders with a expected length', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Addpoint />, div);
  expect(div.querySelectorAll('input')).toHaveLength(1);
});
it('renders with a expected length', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Addpoint type="text" />, div);
  expect(div.querySelector('input')).toHaveAttribute('type', 'text');
});
/* test('should increment to 2 and subtotal increases', () => {
  const { getByTestId } = render(<Addpoint />);
  fireEvent.click(getByTestId('simple'));
  expect(getByTestId('alert')).toHaveTextContent('gv');
}); */
