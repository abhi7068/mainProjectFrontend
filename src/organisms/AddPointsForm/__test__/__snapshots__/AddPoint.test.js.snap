// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
    novalidate=""
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink="false"
      >
        Add Amount
        <span
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
          â€‰*
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiInput-input"
          name="points"
          required=""
          type="text"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root Mui-required"
      />
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
      data-testid="simple"
      style="float: right; background-color: skyblue; margin-top: 36px;"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Add Points
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
  <div
    class="MuiGrid-root MuiGrid-container"
  />
</DocumentFragment>
`;
